Stratis.Bitcoin/Base/BaseFeature.cs

```diff
@@ -117,7 +134,10 @@ public override void Start()
             connectionParameters.IsRelay = !this.nodeSettings.ConfigReader.GetOrDefault("blocksonly", false);
             connectionParameters.TemplateBehaviors.Add(new ChainHeadersBehavior(this.chain, this.chainState));
             connectionParameters.TemplateBehaviors.Add(new AddressManagerBehavior(this.addressManager));
+            var timeSyncBehaviorState = new TimeSyncBehaviorState(this.dateTimeProvider, this.nodeLifetime, this.asyncLoopFactory, this.loggerFactory);
```

Comment:
Just a suggestion you can let TimeSyncBehaviorState be injected in the constructor of the feature.

Reply:
Do you mean to treat the TSBState just like chainRepository in this code? I.e. call the constructor inside `UseBaseFeature`?

Reply:
Almost, you dont need to call the constructor just register the type tot he DI and it will be done for you.